Feature/rewrite example json generator

dipeshsingh253 opened the pull request:
Hi @timonback ,

I've updated ExampleJsonGenerator.java to return JsonNode in internal methods. Initially, I attempted with ObjectNode but it had limitations, allowing only key-value pair objects. JsonNode offers more flexibility with Jackson node classes.

Let me know if any modifications are needed.

Thanks!

timonback commented:
Hi @dipeshsingh253 ,

Based on a brief look this looks good, we will do a review on Friday.

I started the pipeline, you probably encounter the spotless check that we configured to forbid wildcard imports. In case there are other formatting things, spotlessApply can fix most of them automatically.

Great to see your effort! I am positive that we can resolve https://github.com/springwolf/springwolf-core/issues/368.

dipeshsingh253 commented:
> Hi @dipeshsingh253 ,
> 
> Based on a brief look this looks good, we will do a review on Friday.
> 
> I started the pipeline, you probably encounter the spotless check that we configured to forbid wildcard imports. In case there are other formatting things, spotlessApply can fix most of them automatically.
> 
> Great to see your effort! I am positive that we can resolve #368.

Hey @timonback , I hope you are doing well. This is just a reminder text regarding the changes to the preview.

Thanks !!!

dipeshsingh253 commented:
Hi @timonback , @sam0r040 , this time I have pushed the changes from my home setup and I actually used a script to push the changes in my home setup, and by default, I used ```git add .``` command to add changes and now as you can see there are 70 file changes. I have searched and found that git revert can resolve this but I am not sure because I have added these changes in my last commit using ```git commit --amend```. Any advice for me on this.?

timonback commented:
Hi @dipeshsingh253 
Yes, `git reset` is probably the command you want to use. Check how other people resolved it: https://stackoverflow.com/questions/927358/how-do-i-undo-the-most-recent-local-commits-in-git#927386

Use git status to verify what you are committing

dipeshsingh253 commented:
Hi @timonback , Thanks it worked out in the end, also I have added a .gitignore file as well in this commit please review it and if not needed I can definitely rollback it to its original state, I have just added this because this can avoid situations like this one.

Thanks !!! Looking forward to this.

timonback commented on .gitignore:
```diff
@@ -19,3 +19,82 @@ springwolf-plugins/springwolf-sqs-plugin/build/
 .idea/
 /gradle.properties
 
+# Eclipse IDE
```
I like the eclipse comment and the improvement for Eclipse users.

Are L23-25 could be sufficient? Are the remaining lines needed as well? 

Based on https://labs.consol.de/de/development/git/2017/02/22/gitignore.html I assume that those lines should match recursively.

sam0r040 commented:
> Good morning, @timonback @sam0r040 ,
> 
> Are there any additional modifications required, or can we proceed with this PR?

Please have look at the failing builds. I will have a look at this PR on Friday.

dipeshsingh253 commented:
> > Good morning, @timonback @sam0r040 ,
> > Are there any additional modifications required, or can we proceed with this PR?
> 
> Please have look at the failing builds. I will have a look at this PR on Friday.

They are issues related to imports. In my local Windows setup somehow IntelliJ Idea keeps removing some dependencies even when they are used.

sam0r040 commented on springwolf-core/src/main/java/io/github/stavshamir/springwolf/schemas/example/ExampleJsonGenerator.java:
```diff
@@ -61,15 +57,17 @@ public Object fromSchema(Schema schema, Map<String, Schema> definitions) {
 
     static String buildSchema(Schema schema, Map<String, Schema> definitions) {
-        return buildSchemaInternal(schema, definitions, new HashSet<>());
+        return buildSchemaInternal(schema, definitions, new HashSet<>()).toString();
```
As this is the only thing that is missing and I am already planning a quick refactoring, I will resolve this without being changed.

sam0r040 commented:
I rebased the change on master and fixed the imports, I think this one is done.

dipeshsingh253 commented:
Thanks @sam0r040 and @timonback . This was my first open-source contribution. Thanks for all your assistance, it was really great working on this issue.

sam0r040 commented:
Very welcome, thanks a lot for your contribution!
